src/partition.js

```diff
@@ -1,30 +1,29 @@
-var _curry2 = require('./internal/_curry2');
-var _reduce = require('./internal/_reduce');
+var filter = require('./filter');
+var juxt = require('./juxt');
+var reject = require('./reject');
 
 
 /**
- * Takes a predicate and a list and returns the pair of lists of elements which
- * do and do not satisfy the predicate, respectively.
+ * Takes a predicate and a list or other "filterable" object and returns the
+ * pair of filterable objects of the same type of elements which do and do not
+ * satisfy, the predicate, respectively.
  *
  * @func
  * @memberOf R
  * @since v0.1.4
  * @category List
- * @sig (a -> Boolean) -> [a] -> [[a],[a]]
- * @param {Function} pred A predicate to determine which array the element belongs to.
- * @param {Array} list The array to partition.
- * @return {Array} A nested array, containing first an array of elements that satisfied the predicate,
- *         and second an array of elements that did not satisfy.
+ * @sig Filterable f => (a -> Boolean) -> f -> [f,f]
```

Comment:
I don't think `Pair` makes much sense here, as we will always actually return a 2-element array from this, correct?